jupyterlab_vega/__init__.py

```diff
@@ -26,13 +26,24 @@ def _jupyter_nbextension_paths():
 #   from jupyterlab_vega import Vega
 #   Vega(data)
 
-def Vega(data):
-    bundle = {
-        'application/vnd.vega.v2+json': data,
-        'application/json': data,
-        'text/plain': '<jupyterlab_vega.Vega object>'
-    }
-    display(bundle, raw=True)
+class Vega(JSON):
+
+    @property
+    def data(self):
+        return self._data
+    
+    @data.setter
+    def data(self, data):
+        if isinstance(data, str):
+            data = json.loads(data)
+        self._data = data
+
+    def _ipython_display_(self):
+        bundle = {
+            'application/vnd.vega.v2+json': self.data,
+            'text/plain': '<jupyterlab_geojson.GeoJSON object>'
+        }
+        display(bundle, raw=True)
 
 def VegaLite(spec, data):
```

Comment:
Yeah, our object will probably need a slight different set of args. We probably want:

* First arg of `spec` that could eithe rhave the data embedded or not
* Keyword argument of `data` that can only be a DataFrame
* Keyword argument for `filename` that would be the full filename of the spec+data
* Keyword argument for `url` that would be the full URL for the spec+data

We may need to override more of the logic from the base `DisplayObject` because we are effectivly renaming `data->spec`. It may be easier to just copy source code over and modify it.

Did you see the thread on Altair that we want to move the utils file and the display objects over to Altair? It is fine to finish them up here and then move them though.